301-create-ad-forest-with-subdomain/azuredeploy.json

```diff
@@ -0,0 +1,711 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "createChildDomain": {
+      "type": "bool",
+      "metadata": {
+        "description": "Do you want a child domain? If false, the parameters for the child domain are ignored. "
+      },
+      "defaultValue": true
+    },
+    "createSecondDc": {
+      "type": "bool",
+      "metadata": {
+        "description": "Do you want a second DC in each domain? If false: parameters for DC2 and DC4 are ignored. "
+      },
+      "defaultValue": false
+    },
+    "DomainName": {
+      "type": "string",
+      "metadata": {
+        "description": "Full FQDN name for the forest root domain."
+      },
+      "defaultValue": "contoso.com",
+      "minlength": 4
+    },
+    "ChildDomainName": {
+      "type": "string",
+      "metadata": {
+        "description": "SHORT name for the child domain. New AD trees are not allowed."
+      },
+      "defaultValue": "child",
+      "minlength": 2,
+      "maxlength": 15
+    },
+
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the admin account for the Domain(s)"
+      },
+      "defaultValue": "adAdministrator",
+      "minlength": 6,
+      "maxlength": 15
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The (complex!) password for the Administrator account of the new VMs and Domain(s)"
+      },
+      "minlength": 8
+    },
+
+    "VMSize": {
+      "type": "string",
+      "allowedValues": [

```

Comment:
remove allowed values unless there is a specific req for the size

Reply:
There _is_ a dependency on size. Anything smaller does not deploy reliably (too slow), anything larger has no point. Also, in a subordinate template I parse the VM type name to see what kind of managed disk (Standard/Premium) is needed.

Still want to me to take the allowed values out?